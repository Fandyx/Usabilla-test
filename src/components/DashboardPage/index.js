import React, { Component } from 'react';
import Dashboard from './components/Dashboard';
import Filters from './components/Filters';
import styles from './styles/styles'
class DashboardPage extends Component {
    state = {
        searchText:'',
        ratings:[1,2,3,4,5]
    }
    setSearchText = (searchText) => {
        this.setState({searchText})
    }
    setRatings = (ratings) => {
        this.setState({ratings})
    }
    render() {
        return (
            <div style={styles.dashboardStyle}>
                <Filters setSearchText={this.setSearchText} setRatings={this.setRatings}/>
                <Dashboard {...this.props} {...this.state} />
            </div>
        );
    }
}

export default DashboardPage;