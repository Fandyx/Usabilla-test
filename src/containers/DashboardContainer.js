import React from 'react';
import PropTypes from 'prop-types';
import DashboardPage from "../components/DashboardPage"
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import * as actions from '../actions/reviews';

export class Reviews extends React.Component
{
      componentDidMount() {
        this.props.actions.fetchReviews()
      }  

      render(){
        return (
          <div>
            <DashboardPage {...this.props}/>
          </div>
        );
      }
}

Reviews.propTypes = {
    actions: PropTypes.object.isRequired,
    reviews: PropTypes.object.isRequired
  };

function mapStateToProps(state) {
    return {
      reviews: state.reviews
    };
  }
  
  function mapDispatchToProps(dispatch) {
    return {
      actions: bindActionCreators(actions, dispatch)
    };
  }
  
  export default connect(
    mapStateToProps,
    mapDispatchToProps
  )(Reviews);