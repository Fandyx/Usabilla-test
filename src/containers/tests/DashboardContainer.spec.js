import React from "react";
import { shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { create } from "react-test-renderer";
import { DashboardContainer } from "../DashboardContainer";
import DashboardPage from "../../components/DashboardPage/index";

describe("<DashboardContainer />", () => {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const initialState = {
    reviews: {
      list: [],
      isLoading: false,
      count: 0
    }
  };
  let store, wrapper;
  beforeEach(() => {
    store = mockStore(initialState);

    wrapper = shallow(
      <DashboardContainer
        store={store}
        reviews={initialState.reviews}
        actions={{ fetchReviews: jest.fn() }}
      />
    );
  });

  it("should contain <DashboardPage />", () => {
    expect(wrapper.find(DashboardPage).length).toEqual(1);
  });

  it("should match snapshot", () => {
    const component = create(
      <Provider store={store}>
        <DashboardContainer
          reviews={initialState.reviews}
          actions={{ fetchReviews: jest.fn() }}
        />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
