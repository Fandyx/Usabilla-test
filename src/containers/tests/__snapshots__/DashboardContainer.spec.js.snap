// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardContainer /> should match snapshot 1`] = `
<div>
  <div>
    <nav
      style={
        Object {
          "background": "white",
          "boxShadow": "1px 1px 5px #ccc",
          "marginBottom": "15px",
          "padding": "15px 0",
          "textAlign": "center",
        }
      }
    >
      <span
        className="glyphicon glyphicon-dashboard"
      />
      <h2
        style={
          Object {
            "display": "inline-block",
            "fontSize": "16px",
            "fontWeight": "400",
            "letterSpacing": "1px",
            "margin": "0",
            "marginLeft": "10px",
          }
        }
      >
        Dashboard
      </h2>
    </nav>
    <div
      style={
        Object {
          "margin": "auto",
          "width": "85%",
        }
      }
    >
      <div>
        <input
          className="responsive-input"
          onChange={[Function]}
          placeholder="Search here!"
          style={
            Object {
              "border": "1px solid #ccc",
              "borderRadius": "5px",
              "fontSize": "14px",
              "padding": "3px 12px",
              "width": "135px",
            }
          }
          type="text"
          value=""
        />
        <div
          className="responsive-container"
          style={
            Object {
              "display": "inline-block",
              "marginLeft": "50px",
            }
          }
        >
          <div
            className="ratings active"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "border": "1px solid #ccc",
                "borderRadius": "100px",
                "color": "#ccc",
                "display": "inline-block",
                "fontSize": "14px",
                "fontWeight": 500,
                "height": "30px",
                "lineHeight": "2.1",
                "textAlign": "center",
                "width": "30px",
              }
            }
          >
            1
          </div>
          <div
            className="ratings active"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "border": "1px solid #ccc",
                "borderRadius": "100px",
                "color": "#ccc",
                "display": "inline-block",
                "fontSize": "14px",
                "fontWeight": 500,
                "height": "30px",
                "lineHeight": "2.1",
                "textAlign": "center",
                "width": "30px",
              }
            }
          >
            2
          </div>
          <div
            className="ratings active"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "border": "1px solid #ccc",
                "borderRadius": "100px",
                "color": "#ccc",
                "display": "inline-block",
                "fontSize": "14px",
                "fontWeight": 500,
                "height": "30px",
                "lineHeight": "2.1",
                "textAlign": "center",
                "width": "30px",
              }
            }
          >
            3
          </div>
          <div
            className="ratings active"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "border": "1px solid #ccc",
                "borderRadius": "100px",
                "color": "#ccc",
                "display": "inline-block",
                "fontSize": "14px",
                "fontWeight": 500,
                "height": "30px",
                "lineHeight": "2.1",
                "textAlign": "center",
                "width": "30px",
              }
            }
          >
            4
          </div>
          <div
            className="ratings active"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "transparent",
                "border": "1px solid #ccc",
                "borderRadius": "100px",
                "color": "#ccc",
                "display": "inline-block",
                "fontSize": "14px",
                "fontWeight": 500,
                "height": "30px",
                "lineHeight": "2.1",
                "textAlign": "center",
                "width": "30px",
              }
            }
          >
            5
          </div>
        </div>
      </div>
      <div>
        <div
          style={
            Object {
              "border": "1px solid transparent",
              "borderRadius": "38px 38px 5px 5px",
              "overflow": "hidden",
            }
          }
        >
          <table
            cellPadding="10"
            cellSpacing="0"
            style={
              Object {
                "backgroundColor": "white",
                "borderRadius": "5px",
                "boxShadow": "1px 2px 2px #ccc",
                "marginTop": "25px",
                "width": "100%",
              }
            }
          >
            <thead
              style={
                Object {
                  "backgroundColor": "#59636C",
                  "color": "white",
                  "fontWeight": "200",
                  "textAlign": "left",
                }
              }
            >
              <tr>
                <th
                  className="center"
                  style={
                    Object {
                      "fontSize": "14px",
                      "fontWeight": "normal",
                      "padding": "15px 0px",
                    }
                  }
                >
                  Rating
                </th>
                <th
                  className="padding-comment"
                  style={
                    Object {
                      "fontSize": "14px",
                      "fontWeight": "normal",
                      "padding": "15px 0px",
                    }
                  }
                >
                  Comment
                </th>
                <th
                  className="center"
                  style={
                    Object {
                      "fontSize": "14px",
                      "fontWeight": "normal",
                      "padding": "15px 0px",
                    }
                  }
                >
                  Browser
                </th>
                <th
                  className="center"
                  style={
                    Object {
                      "fontSize": "14px",
                      "fontWeight": "normal",
                      "padding": "15px 0px",
                    }
                  }
                >
                  Device
                </th>
                <th
                  className="center"
                  style={
                    Object {
                      "fontSize": "14px",
                      "fontWeight": "normal",
                      "padding": "15px 0px",
                    }
                  }
                >
                  Platform
                </th>
                <th
                  className="center"
                  style={
                    Object {
                      "fontSize": "14px",
                      "fontWeight": "normal",
                      "padding": "15px 0px",
                    }
                  }
                >
                  Details
                </th>
              </tr>
            </thead>
            <tbody
              style={
                Object {
                  "margin": "auto",
                }
              }
            >
              <tr>
                <td
                  className="no-data"
                >
                  No data to match your filters
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
